rclcpp/src/rclcpp/parameter_map.cpp

```diff
@@ -62,7 +64,24 @@ rclcpp::parameter_map_from(const rcl_params_t * const c_params)
         throw InvalidParametersException(message);
       }
       const rcl_variant_t * const c_param_value = &(c_params_node->parameter_values[p]);
-      params_node.emplace_back(c_param_name, parameter_value_from(c_param_value));
+      params[c_param_name].first = Parameter(c_param_name, parameter_value_from(c_param_value));
+    }
+
+    const rcl_node_params_descriptors_t * const c_param_descriptors_node =
+      &(c_params->descriptors[n]);
+    for (size_t p = 0; p < c_param_descriptors_node->num_params; ++p) {
+      const char * const c_param_name = c_param_descriptors_node->parameter_names[p];
+      if (NULL == c_param_name) {
+        std::string message(
+          "At node " + std::to_string(n) + " parameter " + std::to_string(p) + " name is NULL");
```

Comment:
@bpwilcox do we need an auxiliary variable?

Reply:
The `message` string? Probably not needed, this was left from the previous code, not my additions.